api/api/routers/mcp/_mcp_service.py

```diff
@@ -341,34 +342,21 @@ async def list_agents(
         page: int,
         sort_by: AgentSortField,
         order: SortOrder,
-        agent_id: str | None = None,
-        stats_from_date: str = "",
-        with_schemas: bool = False,
     ) -> PaginatedMCPToolReturn[None, AgentResponse]:
         """List all agents with their statistics."""
         try:
-            # Parse from_date or use default
-            parsed_from_date = None
-            if stats_from_date:
-                try:
-                    parsed_from_date = datetime.fromisoformat(stats_from_date.replace("Z", "+00:00"))
-                except ValueError:
-                    pass
-
-            if not parsed_from_date:
-                parsed_from_date = datetime.now() - timedelta(days=7)
+            # Use default date (7 days ago)
+            parsed_from_date = datetime.now() - timedelta(days=7)
 
             # Get agent stats
             stats_by_uid: dict[int, TaskRunStorage.AgentRunCount] = {}
             async for stat in self.storage.task_runs.run_count_by_agent_uid(parsed_from_date):
                 stats_by_uid[stat.agent_uid] = stat
 
-            if agent_id:
-                agents = [await tasks.get_task_by_id(self.storage, agent_id, with_schemas=with_schemas)]
-            else:
-                agents = await tasks.list_tasks(self.storage, with_schemas=with_schemas)
+            # Get all agents with basic schema information (always included for consistency)
+            agents = await tasks.list_tasks(self.storage, with_schemas=True)
```

Comment:
```suggestion
            agents = await tasks.list_tasks(self.storage, with_schemas=False)
```

Schemas are no longer used